### PR TITLE
SAML: Okta IdP configuration guide

### PR DESCRIPTION
**Pull request by nfagerlund**
⚠️ This needs expert review before merging! ⚠️

I don't have access to a working Okta account, and I couldn't verify these instructions in real life, so I'm very nervous about them. Zacc didn't remember the whole Okta experience after all this time, so I had to extrapolate these revisions from the screenshots, the Okta docs (which could be better), and a bunch of ambiguous StackOverflow answers. 

But I _think_ I eventually understood everything that was going on, and I tried to explain it in a more thorough way.

### CODE REVIEW

**Comment by nfagerlund** [comment]
Pardon the mass reviewer tagging, I'm honestly not 100% sure who I should pull in on this one.

File: content/source/docs/enterprise/saml/attributes.html.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ The following SAML attributes correspond to properties of a Terraform Enterprise
 
 ## Username
 
-If Username is specified, the system will assign that username instead of an autogenerated username. When the username is already taken or is invalid the value is not updated but login will complete.
+If Username is specified, TFE will assign that username to the user instead of using an automatic name [based on their email address](./login.html). When the username is already taken or is invalid the value is not updated but login will complete.
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/saml/attributes.html.md]: > ...is invalid, login will still complete, and the existing or default value will be used instead.

File: content/source/docs/enterprise/saml/identity-provider-configuration-okta.html.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    - **Name:** `SiteAdmin` (This is the default name for TFE's site admin [attribute][]; the name of this attribute can be changed in [TFE's SAML settings](./configuration.html) if necessary.)
+    - **Name Format** (drop-down): Boolean
+    - **Value:** An [Okta expression](https://developer.okta.com/reference/okta_expression_language/) that will evaluate to `true` for every user who should have site admin permissions, but `false` for any users who should **not** have site admin permissions. The exact expression depends on the user properties you use to manage admin permissions.
+7. Preview the SAML response and make sure it matches your expectations.
+8. Finish configuring the SAML app in Okta, then copy the provided endpoint URLs and certificate to your Terraform Enterprise SAML settings at `https://<TFE_HOSTNAME>/app/admin/saml`. TFE requires a single sign-on URL, a single log-out URL, and a PEM (base64) encoded X.509 certificate.
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/saml/identity-provider-configuration-okta.html.md]: I think we don't require an SLO, strictly.

File: content/source/docs/enterprise/saml/identity-provider-configuration.html.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ Single sign-on setup instructions for specific identity providers (IdP).
 ## Identity Providers
 
 * [OneLogin](./identity-provider-configuration-onelogin.html)
+* [Okta](./identity-provider-configuration-okta.html)
 * [ADFS](./identity-provider-configuration-adfs.html)
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/saml/identity-provider-configuration.html.md]: I wonder if we should re-order these to go alphabetically?

File: content/source/docs/enterprise/saml/identity-provider-configuration-okta.html.md (hunk 2 above)
Comment (nfagerlund) [reply content/source/docs/enterprise/saml/identity-provider-configuration-okta.html.md]: Oh, right... I know we can't use it, but I thought we still required it for some reason. (All these gaps from not having gone through this setup myself... :| )

Figure we should still say we want it for IdPs where it's available, right? So you don't have to do extra shenanigans if we start using it?

File: content/source/docs/enterprise/saml/identity-provider-configuration.html.md (hunk 3 above)
Comment (nfagerlund) [reply content/source/docs/enterprise/saml/identity-provider-configuration.html.md]: I'd rather go w/ order of prevalence, but idk if anyone has a good guess about that. Alphabetical is fine w/ me.

**Comment by nfagerlund** [comment]
OK, we're doin' it! Can always iterate if someone finds something goofy.

**Pull request merged**